File: career-connect-job-portal/jobportal/src/main/java/com/onkar/jobportal/service/ApplicationService.java
package com.onkar.jobportal.service;

import java.util.List;

import com.onkar.jobportal.dto.ApiResponse;
import com.onkar.jobportal.dto.ApplicationRequest;
import com.onkar.jobportal.entity.JobApplication;

public interface ApplicationService {

    ApiResponse applyJob(ApplicationRequest request);

    List<JobApplication> getApplicationsByUser(Long userId);

    List<JobApplication> getApplicationsByJob(Long jobId);

    // Recruiter Applications
    List<JobApplication> getApplicationsForRecruiter(Long recruiterId);

    ApiResponse updateStatus(Long applicationId, String status);

	List<JobApplication> getApplicationsForCompany(Long companyId);

}

File: career-connect-job-portal/job-portal-frontend/src/services/applicationService.js
import API from "../api/axios";

export const applyJob = (application) => {
    return API.post("/application/apply", application);
};

export const getMyApplications = (userId) => {
    return API.get(`/application/user/${userId}`);
};

export const getApplicationsByJob = (jobId) => {
    return API.get(`/application/job/${jobId}`);
};

export const getApplicationsByRecruiter = (recruiterId) =>
    API.get(`/application/recruiter/${recruiterId}`);

export const updateApplicationStatus = (id, status) => {
    return API.put(`/application/${id}/status?status=${status}`);
};